components/reddit/sources/new-hot-post-on-a-subreddit/new-hot-post-on-a-subreddit.js

```diff
@@ -0,0 +1,184 @@
+const reddit = require("../../reddit.app.js");
+
+module.exports = {
+  key: "new-hot-spot-on-a-subreddit",
+  name: "New hot spot on a subreddit",
+  description:
+    "Emits an event each time a new host post is added to the top 10 a subreddit.",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
```

Comment:
High-level comment: are all of these props required, or are some optional?

Reply:
only subreddit (The subreddit you'd like to watch for new links) prop is required. the other props are optional

Reply:
Ok thanks. Can you set `optional: true` on these props?

Reply:
Sure, added.